File: js/BinaryTreeLevelOrderTraversal.js
var levelOrder = function(root) {
    if(root == null){
        return [];
    }
    let nodes = [];
    nodes.push(root);
    let result = [];
    while(nodes.length > 0){
        let current_level = [];
        let next_level = [];
        nodes.forEach(function(n){
            current_level.push(n.val);
            if(n.left != null){
                next_level.push(n.left);
            }
            if(n.right != null){
                next_level.push(n.right);
            }
        });
        result.push(current_level);
        nodes = next_level;
    }
    return result;
};

File: csharp/BinaryTreeLevelOrderTraversal.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LeetCode
{
    class BinaryTreeLevelOrderTraversal
    {
        public IList<IList<int>> LevelOrder(TreeNode root)
        {
            if (root == null)
            {
                return new List<IList<int>>();
            }
            List<IList<int>> result = new List<IList<int>>();
            List<TreeNode> nodes = new List<TreeNode>();
            nodes.Add(root);
            while (nodes.Any())
            {
                List<int> currentLevel = new List<int>();
                List<TreeNode> nextLevel = new List<TreeNode>();
                foreach (var n in nodes)
                {
                    currentLevel.Add(n.val);
                    if (n.left != null)
                    {
                        nextLevel.Add(n.left);
                    }
                    if (n.right != null)
                    {
                        nextLevel.Add(n.right);
                    }
                }
                result.Add(currentLevel);
                nodes = nextLevel;
            }
            return result;
        }
    }
}
